### PR TITLE
Order-agnostic comparison in TestEnableHAControllerConfigConstraints

### PR DESCRIPTION
**Pull request by manadart**
## Description of change

Fixes intermittent test failures due to ordering of space names inside constraints.Value instance.

## QA steps

Test consistently passes when run in a loop many times.

## Documentation changes

None.

## Bug reference

N/A

### CODE REVIEW

File: apiserver/facades/client/highavailability/highavailability_test.go [hunk 1]
```diff
@@ -223,15 +224,26 @@ func (s *clientSuite) TestEnableHAControllerConfigConstraints(c *gc.C) {
...
-		constraints.MustParse("spaces=random-space,ha-space"),
-	}
+
+	expectedSpaceCons := []string{"ha-space", "random-space"}
 	for i, m := range machines {
 		cons, err := m.Constraints()
 		c.Assert(err, jc.ErrorIsNil)
-		c.Check(cons, gc.DeepEquals, expectedCons[i])
+
+		if i == 0 {
```
Comment (ExternalReality) [inline apiserver/facades/client/highavailability/highavailability_test.go]: Perhaps a comment here to specify why the zero index is interesting?

**Comment by manadart** [comment]
Backed this up and sorted in Config.AsSpaceConstraints as suggested by @jameinel.

**Comment by manadart** [comment]
$$merge$$

**Comment by manadart** [comment]
!!build!!

**Pull request closed without merging**